### DISCUSSION TITLE
Disable Copilot in Markdown files in PhpStorm

### DISCUSSION THREAD

**Discussion by iandunn** [community/community#47578] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

The PhpStorm plugin has preferences to disable Copilot in various file types, but Markdown is missing from that list. 

![Screenshot 2023-02-16 at 7 46 55 AM](https://user-images.githubusercontent.com/484068/219416519-79fb074d-ff87-4bd7-9354-b0fdf32c1d62.png)

Copilot is always getting in the way when I try to edit MD files, in a similar way to comments (#8062). I want to turn it off completely for Markdown.

**Comment by unknowVariable** [comment]
If you're using the Copilot plugin in PhpStorm and you're experiencing issues when editing Markdown files, you can try disabling the plugin for that file type. Although Markdown is not currently on the list of file types in the plugin preferences, you may be able to add it manually.

Here are the steps you can follow to disable Copilot for Markdown files:

1. In PhpStorm, go to Preferences (or Settings on Windows) by selecting PhpStorm > Preferences from the menu bar.

2. In the Preferences window, click on "Editor" in the left-hand sidebar, and then select "Code Style."

3. Scroll down to the "File types" section and click on the plus (+) button to add a new file type.

4. In the "Add File Type" dialog box, type "Markdown" as the file type name and "md" as the file extension. Then click "OK" to add the new file type.

5. In the "File Types" section, click on "Markdown" to select it. Then scroll down to the "Code Generation" section and uncheck "Enable Copilot."

6. Click "Apply" and "OK" to save your changes.

After completing these steps, Copilot should be disabled for Markdown files in PhpStorm. If you still experience issues or have any questions, I recommend reaching out to the PhpStorm support team for further assistance.

**Reply by iandunn** [reply]
There are several things that are fundamentally wrong with that; did you copy/paste it from ChatGPT?

If so, please don't do that, [it's very unhelpful](https://stackoverflow.com/help/gpt-policy).

**Comment by SilasRodrigues19** [comment]
Hey @iandunn 

Press CTRL Shift P and open your `settings.json`file, then copy this code into your `json`:

```json
"github.copilot.enable": {
    "markdown": false,
  },
```

**Reply by iandunn** [reply]
Hey 👋🏻  I tried that, but it looks like I don't have `ctr shift p` bound to anything. I don't see a `settings.json` anywhere in `~/Library/ApplicationSupport/JetBrains/PhpStorm2022.3` either.

Can you let me know the menu location, or name of the action in the `keymap` preferences?

**Reply by iandunn** [reply]
Oh, it looks like you're talking about VSCode? This post is about the PhpStorm plugin.

**Reply by SilasRodrigues19** [reply]
Sorry.

Try to see if there is any option similar to this one in PHPStorm, it may have some option inside the file to enable or disable that specific extension.

![image](https://user-images.githubusercontent.com/47532703/219226879-65655ad5-56c7-4f1b-a2f9-c239c9891d21.png)

**Reply by iandunn** [reply]
No worries. I haven't seen any dialogs like that from the PhpStorm plugin.

It does have a settings screen with a list of languages to enable/disable, but Markdown is missing from the list.

**Comment by RichStone** [comment]
Copilot in markdown is real trouble :D

By today, I'm seeing markdown as an option in the "disabled language" list (RubyMine).
